benchmarks/nemo/nemolite2d/manual_versions/psykal_cpp_ompss/time_step_ompss.cpp

```diff
@@ -0,0 +1,274 @@
+                area_v, gphiv, rdt, cbfr, visc, omega, d2r, g);
+        }
+      }
+    }
+
+    // Boundary conditions bc_ssh kernel (internal domain)
+    for(int jj = internal_ystart; jj <= internal_ystop; jj+=TASK_SIZE){
+      int stop = std::min(jj+TASK_SIZE-1, internal_ystop);
+      int task_row = jj/TASK_SIZE;
+      #pragma omp task firstprivate(jj, task_row, stop) out( ssha_t[task_row] )
+      for( int task_j =jj; task_j <= stop; task_j++){
+        for(int ji = internal_xstart; ji <= internal_xstop; ji++){
+            bc_ssh_code(ji, task_j, width, istep, ssha_t, tmask, rdt);
+        }
+      }
+    }
+
+    // Boundary conditions bc_solid_u kernel (whole domain but top x boundary)
+    for(int jj = internal_ystart - 1; jj <= internal_ystop + 1; jj+=TASK_SIZE){
+      int stop = std::min(jj+TASK_SIZE-1, internal_ystop+1);
+      int task_row = jj/TASK_SIZE;
+      #pragma omp task firstprivate(jj, task_row, stop) out( ua[task_row] )
+      for( int task_j =jj; task_j <= stop; task_j++){
+        for(int ji = internal_xstart - 1; ji <= internal_xstop; ji++){
+            bc_solid_u_code(ji, task_j, width, ua, tmask);
+        }
+      }
+    }
+
+    // Boundary conditions bc_solid_v kernel (whole domain but top y boundary)
+    for(int jj = internal_ystart - 1; jj <= internal_ystop; jj+=TASK_SIZE){
+      int stop = std::min(jj+TASK_SIZE-1, internal_ystop);
+      int task_row = jj/TASK_SIZE;
+      #pragma omp task firstprivate(jj, task_row, stop) out( ua[task_row] )
+      for( int task_j =jj; task_j <= stop; task_j++){
+        for(int ji = internal_xstart - 1; ji <= internal_xstop + 1; ji++){
+            bc_solid_v_code(ji, task_j, width, va, tmask);
+        }
+      }
+    }
+    // Boundary conditions bc_flather_u kernel (whole domain but top x boundary)
+    for(int jj = internal_ystart - 1; jj <= internal_ystop + 1; jj+=TASK_SIZE){
+      int stop = std::min(jj+TASK_SIZE-1, internal_ystop);
+      int task_row = jj/TASK_SIZE;
+      #pragma omp task firstprivate(jj, task_row, stop) out( ua[task_row] ) \
+                                                         in( ua[task_row] , \
+                                                             ua[task_row-1], \
+                                                             ua[task_row+1], \
+                                                             hu[task_row], \
+                                                             sshn_u[task_row], \
+                                                             sshn_u[task_row+1], \
+                                                             sshn_u[task_row-1] )
+               
+                                                                      
+      for( int task_j =jj; task_j <= stop; task_j++){
+        for(int ji = internal_xstart - 1; ji <= internal_xstop; ji++){
+            bc_flather_u_code(ji, task_j, width, ua, hu, sshn_u, tmask, g);
+        }
+      }
+    }
+
+#pragma omp taskwait
```

Comment:
Please add a comment explaining why this wait is required.

Reply:
Done